cloudify_vsphere/contentlibrary/__init__.py

```diff
@@ -22,24 +22,33 @@ class ContentLibrary(object):
 
     def __init__(self, connection_config):
         self.config = connection_config
-        response = requests.request(
+        # we need set it empty for correct delete
+        self.session_id = None
+        value = self._call(
             "POST", "https://{host}/rest/com/vmware/cis/session"
                     .format(host=self.config['host']),
             auth=(self.config['username'], self.config['password']),
             verify=not self.config.get('allow_insecure', False))
-        response.raise_for_status()
-        self.session_id = response.cookies['vmware-api-session-id']
-        json_value = response.json()
-        if self.session_id != json_value.get('value'):
+        if self.session_id != value:
```

Comment:
This looks like it can only ever pass if the call returns a null value?

Reply:
Content library return cookie in cookies and {"value": "cookie"}. And _call has side effect - it always update cookie from response cookies.